crates/runtime/src/lib.rs

```diff
@@ -12,14 +14,74 @@ mod memory;
 pub mod store;
 
 pub use constraint::Constraint;
+use errors::{FunctionCallError, VMRuntimeError};
+use logic::{Outcome, VMContext, VMLimits, VMLogic, VMLogicError};
+use memory::WasmerTunables;
+use store::Storage;
 
 pub type RuntimeResult<T, E = VMRuntimeError> = Result<T, E>;
 
+#[derive(Message, Debug)]
+#[rtype(result = "RuntimeResult<Outcome>")]
+pub struct ExecuteMsg {
+    pub blob: Vec<u8>,
+    pub method_name: String,
+    pub context: VMContext,
+    pub storage: Box<dyn Storage + Send>,
+}
+#[derive(Debug)]
+pub struct RuntimeManager {
+    pub tasks: BTreeMap<ContextId, Arc<Mutex<()>>>,
+    pub limits: VMLimits,
+}
+
+impl Actor for RuntimeManager {
+    type Context = Context<Self>;
+}
+
+impl RuntimeManager {
+    pub fn new(limits: VMLimits) -> Self {
+        RuntimeManager {
+            tasks: BTreeMap::new(),
+            limits,
+        }
+    }
+}
+
+impl Handler<ExecuteMsg> for RuntimeManager {
+    type Result = ResponseFuture<RuntimeResult<Outcome>>;
+
+    fn handle(&mut self, msg: ExecuteMsg, _ctx: &mut Self::Context) -> Self::Result {
+        let mutex = self
+            .tasks
+            .entry(msg.context.context_id.into())
+            .or_default()
+            .clone();
+
+        let storage = msg.storage;
+
+        let limits = self.limits.clone();
+
+        let future = async move {
+            let _lock = mutex.lock().await;
+
+            let handle = global_runtime().spawn_blocking(move || {
+                let result = run(&msg.blob, &msg.method_name, msg.context, storage, &limits);
+                result
+            });
+
+            handle.await.unwrap()
+        };
+
+        Box::pin(future)
+    }
+}
+
 pub fn run(
     code: &[u8],
     method_name: &str,
     context: VMContext,
-    storage: &mut dyn Storage,
+    storage: Box<dyn Storage>,
```

Comment:
you shouldn't need this change, we only need to `Box` because of the `'static` requirement of [`Addr::send`](https://docs.rs/actix/latest/actix/struct.Addr.html#method.send)